### ISSUE TITLE
Replace usage of Lodash `cloneDeep` with native `structuredClone`.

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#6492]
## Feature Description

As discussed in [this article](https://www.builder.io/blog/structured-clone), the native [`structuredClone`](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) function can be used instead of `cloneDeep`. It has good browser support and would allow us to marginally reduce the bundle size.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Use of `cloneDeep` within the codebase should be replaced with `structuredClone`.

## Implementation Brief

* Wait until we upgrade our version of Node to at least `17` (probably `18`), see #6026.
* Replace every import of `cloneDeep` with `structuredClone`, eg `cloneDeep( object )` to `structuredClone( object )`.
* At the time of writing, we have usage in the following files:
  * `assets/js/modules/analytics-4/datastore/properties.test.js`
  * `assets/js/modules/analytics-4/components/module/ModulePopularPagesWidgetGA4/index.js`
  * `assets/js/modules/analytics-4/components/dashboard/DashboardAllTrafficWidgetGA4/UserCountGraph.js`
  * `assets/js/modules/analytics-4/components/dashboard/DashboardAllTrafficWidgetGA4/UserDimensionsPieChart.js`
  * `assets/js/modules/analytics-4/utils/data-mock.js`
  * `assets/js/modules/reader-revenue-manager/components/settings/SettingsEdit.stories.js`
  * `assets/js/modules/reader-revenue-manager/datastore/publications.test.js`
  * `assets/js/components/GoogleChart/utils.js`
  * `assets/js/components/wp-dashboard/WPDashboardPopularPagesGA4.js`
  * `storybook/utils/resetGlobals.js`

### Test Coverage

* Tests should continue to pass without change; no new tests are needed.

## QA Brief

* 

## Changelog entry

*

**Comment by tofumatt** [comment]
ACs 👍🏻, moving to IB.

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by techanvil** [comment]
Note, while working on #6216 I noticed that `structuredClone` is not available in Node until version 17.

Given that we're currently using Node 14, the IB should be reevaluated with this in mind.

**Comment by tofumatt** [comment]
Rather than add what is essentially another workaround for `structuredClone` not being available, I'd rather this be blocked by an upgrade to the latest Node LTS, which is `node` `18`.

We already have an issue for updating node (#6026), which was filed about 4 months ago. At the time LTS was 16, but I think it's worth exploring an upgrade to 18 as some other updates depend on a newer version as well.

Then we can implement this according to the IB. I've added one note to the IB mentioning this requires Node 17+. I've flagged the node upgrade as blocking this issue as well, does that sound good @techanvil?

**Comment by techanvil** [comment]
Thanks @tofumatt, that sounds good to me.

IB :white_check_mark:

**Comment by techanvil** [comment]
Thanks, @zutigrm, good shout. I've amended the list of files in the IB to reflect the current usage, and with no other material change needed to the previously approved IB I'll move this straight to the EB.

**Comment by ankitrox** [comment]
@tofumatt It looks like even though we have updated the node environment to 22, the tests are run in jsdom which does not support all browser APIs including `structuredClone`.

I also found one related issue in jsdom repository: https://github.com/jsdom/jsdom/issues/3363

I have added the polyfill as of now which we can remove (not necessary even if `structuredClone` is available in jsdom) later. Please let me know your thoughts on this.

Another option is to close this issue for now and revisit when the above mentioned jsdom issue is addressed, because adding a polyfill would not serve any purpose and we can continue to use `cloneDeep` as is anyways.

CC: @techanvil

**Comment by tofumatt** [comment]
I think it's fine to use the polyfill in that case; let's just leave a comment next to it and link to the JSDom issue that's tracking it so we can know when to remove the polyfill 🙂

**Comment by ankitrox** [comment]
Marking this issue as blocked by #5862 and will be waiting for it to get merged.
Moving this to EB, but keeping this assigned to myself.

@ivonac4 Can we unassign this issue from the sprint as of now?

**Comment by techanvil** [comment]
As #5862 has effectively been superseded by #11436 (see conversation in [Slack](https://fueled.slack.com/archives/C077DB69US2/p1767782737992749)), I've updated the dependencies for this issue accordingly.
